prepare_input_files_lists/list_dataset_files.py

```diff
@@ -56,8 +62,26 @@ def __get_files_list_from_info_dict(info_dict):
     bash_command = f"xrdfs {redirector} ls {path} | grep -E \"{regex}\" | sort -n"
     files_list = __run_bash_command(bash_command).split("\n")
     files_list = [f"{redirector}{file_name}" for file_name in files_list]
-    files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("_")[0]))
-
+    #if log is in file name, remove it
+    print(path)
+    files_list = [file_name for file_name in files_list if "log" not in file_name]
+    files_list = [file_name for file_name in files_list if "test.sh" not in file_name]
+    #check if nanodata is in file name, then sort by the number after the last underscore
+    print("file: " , files_list[0])
+    if "nano_data" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("_")[2].replace(".root","")))
+    elif "PFNanoSuper" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("-")[2].replace(".root","")))
+    elif "PFNanoAOD_SVJL_" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("-")[6].replace(".root","")))
+    elif "PFNanoAOD_SVJtaus_" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("-")[6].replace(".root","")))
+    elif "PFNANOAOD" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("-")[2].replace(".root","")))
+    elif "PFNanoAOD" in files_list[0].split("/")[-1]:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("-")[2].replace(".root","")))
+    else:
+        files_list = sorted(files_list, key=lambda x: int(x.split("/")[-1].split("_")[0]))
```

Comment:
I don't know why this is needed, it look completely hard coded, this should be in the dataset config, not here.

Reply:
here the code was breaking when just using what was there before ... but yes, it is of course hardcoded, but does not break your things